sensor/kubernetes/fake/virtualmachines.go

```diff
@@ -55,6 +58,23 @@ func validateVMWorkload(workload VirtualMachineWorkload) VirtualMachineWorkload
 		log.Warnf("virtualMachineWorkload.updateInterval not set or <= 0; defaulting to %s", defaultVMUpdateInterval)
 		workload.UpdateInterval = defaultVMUpdateInterval
 	}
+
+	// Sanity check timing relationships
+	minLifecycle := time.Duration(float64(workload.LifecycleDuration) * (1 - lifecycleJitterPercent))
+	if workload.UpdateInterval > minLifecycle {
+		log.Warnf("virtualMachineWorkload.updateInterval (%s) > minimum lifecycle (%s); updates may never occur",
```

Comment:
Can we make this a bit clearer to the user by explaining the relationship between `LifecycleDuration` and `minimum lifecycle`? It is OK when reading this code because of the comment in line 29, but the yaml file does not mention the 50% jitter so this may be a surprise.

Also perhaps something like "updates may occur less often than intended" could be somewhat more useful than ""may never occur" in understanding the potential issue

Reply:
I rewritten this part, added better warnings, expanded tests in [eeeeaeb](https://github.com/stackrox/stackrox/pull/17913/commits/eeeeaeb28a859dd1e6c1adf42eab2c6fd327d5f4) (what a cool git sha)